admin/views.py

```diff
@@ -553,11 +577,19 @@ def resource_edit(resource_type, resource_id):
 
     for attribute in editable_attributes:
         attribute_value = request.form.get(attribute["name"])
-        print('attribute_value....', attribute_value)
-        validated_attribute_value = validate_resource_attribute(resource_type, attribute, attribute_value)
-        print('validated_attribute_value....', validated_attribute_value)
+        # print('attribute_value....', attribute_value)
+        validated_attribute_value = validate_resource_attribute(
+            resource_type, attribute, attribute_value
+        )
+        # print('validated_attribute_value....', validated_attribute_value)
         setattr(resource, attribute["name"], validated_attribute_value)
 
+    if attribute["name"] == admin_configs["user"]["secret"]:
```

Comment:
This is incorrect because `attribute` is a loop variable (defined from `editable_attributes` for the loop) but you're using it outside the loop. Which means that the `attribute` variable will have the value of the last element in the array and your assumption that the last element in attributes will always be the password is incorrect.

The code will break if the "password" (or the config secret) is not the last element of `editable_attributes`.